### PR TITLE
Feature(token): Prechecks on Token Operations

### LINKED ISSUE #60: Add Token Pre-check validations

### PR DESCRIPTION
**Pull request by failfmi**
**Related issue(s)**:
Closes #60 

**Summary of the change**:
- Update `HAPI.html`.
- Add prechecks on Token Operations.
  Validates inputs for missing/improper passed protobuf properties.
- Add Token Transfers Size Limit constraint

**External impacts**:
None.

**Applicable documentation**
- [ ] Javadoc
- [x] HAPI protobuf comments
- [ ] README

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/txns/validation/TokenChecks.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
+package com.hedera.services.txns.validation;
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/txns/validation/TokenChecks.java:1]: You need to add the `Apache Copyright text` at the beginning of the file

File: hedera-node/src/main/java/com/hedera/services/txns/validation/TokenChecks.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+                uniqueAccounts.add(adjustment.getAccountID());
+                net += adjustment.getAmount();
+            }
+
+            if (uniqueAccounts.size() < tokenTransferList.getTransfersCount()) {
+                return ACCOUNT_REPEATED_IN_ACCOUNT_AMOUNTS;
+            }
+
+            if (net != 0) {
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/txns/validation/TokenChecks.java:63]: I am not 100% sure about the sum of net-zero. Let's discuss it.

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]

File: hedera-node/src/main/java/com/hedera/services/legacy/config/PropertiesLoader.java [hunk 3]
```diff
@@ -108,6 +108,8 @@ public static int getTransferAccountListSize() {
 		return SyncPropertiesObject.getTransferListSizeLimit();
 	}
 
+	public static int getTokenTransfersListSize() { return SyncPropertiesObject.getTokenTransfersListSizeLimit(); }
```
[the remark this answers is not in the document]
Comment (failfmi) [reply hedera-node/src/main/java/com/hedera/services/legacy/config/PropertiesLoader.java:111]: Does not really matter imo

File: hedera-node/src/main/java/com/hedera/services/txns/validation/ContextOptionValidator.java [hunk 4]
```diff
@@ -118,6 +151,30 @@ public ResponseCodeEnum queryableTopicStatus(TopicID id, FCMap<MerkleEntityId, M
 				.orElse(INVALID_TOPIC_ID);
 	}
 
+	@Override
+	public ResponseCodeEnum tokenSymbolCheck(String symbol) {
```
[the remark this answers is not in the document]
Comment (failfmi) [reply hedera-node/src/main/java/com/hedera/services/txns/validation/ContextOptionValidator.java:155]: Yes.

File: hedera-node/src/test/java/com/hedera/services/txns/token/TokenTransferTransitionLogicTest.java [hunk 5]
```diff
@@ -124,11 +136,185 @@ public void setsFailInvalidIfUnhandledException() {
...
+	@Test
+	public void acceptsValidTxn() {
+		givenValidTxnCtx();
+
+		// expect:
+		assertEquals(OK, subject.syntaxCheck().apply(tokenTransactTxn));
+	}
+
+	@Test
+	public void rejectsIncorrectTransfersLength() {
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/test/java/com/hedera/services/txns/token/TokenTransferTransitionLogicTest.java:148]: Some of the tests here seem that they should be `TokenListChecks` unit tests

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**